source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -41,10 +41,12 @@ namespace {
 constexpr auto filter_metadata_key = "com.amazonaws.lambda";
 constexpr auto egress_gateway_metadata_key = "egress_gateway";
 
-void setLambdaHeaders(Http::RequestHeaderMap& headers, absl::string_view function_name,
+void setLambdaHeaders(Http::RequestHeaderMap& headers, const absl::optional<Arn>& arn,
                       InvocationMode mode) {
   headers.setMethod(Http::Headers::get().MethodValues.Post);
-  headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", function_name));
+  auto lambda_arn = "arn:" + arn->partition() + ":" + arn->service() + ":" + arn->region() + ":" +
```

Comment:
Can we add a method inside [Arn class](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/aws_lambda/aws_lambda_filter.h#L21) so that we can directly call `arn->value()` so that we get the arn string value entirely? Then we don't have to construct the arn value every time. The config has the entire arn as a string in itself.

Also worth considering if we can simplify `class Arn`. Currently we only use `region()` & `functionName()` method inside. With this PR we can even get rid of `functionName` and other unused functions.

Reply:
Sure. Actually, planning to add another member variable `arn_` in `Arn()` class which retains the original ARN string from the config.

I would keep the other functions since they could be used in future after parsing ARN to extract specific fields.

Reply:
that should work as well.